likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/post/detail/view/LMFeedPostDetailFragment.kt

```diff
@@ -0,0 +1,1403 @@
+        binding.commentComposer.setCommentInputBoxHint(getString(R.string.lm_feed_write_a_comment))
+        commentComposer.setStyle(LMFeedStyleTransformer.postDetailFragmentViewStyle.commentComposerStyle)
+    }
+
+    private fun receiveExtras() {
+        postDetailExtras = LMFeedExtrasUtil.getParcelable(
+            arguments,
+            LM_FEED_POST_DETAIL_EXTRAS,
+            LMFeedPostDetailExtras::class.java
+        ) ?: throw emptyExtrasException(TAG)
+    }
+
+    private fun initUI() {
+        initPostDetailRecyclerView()
+        //todo:
+//        initMemberTaggingView()
+        initSwipeRefreshLayout()
+    }
+
+    private fun initPostDetailRecyclerView() {
+        fetchPostData()
+        binding.rvPostDetails.apply {
+            setAdapter(
+                this@LMFeedPostDetailFragment,
+                this@LMFeedPostDetailFragment,
+                this@LMFeedPostDetailFragment
+            )
+
+            //set scroll listener
+            val paginationScrollListener =
+                object : LMFeedEndlessRecyclerViewScrollListener(linearLayoutManager) {
+                    override fun onLoadMore(currentPage: Int) {
+                        if (currentPage > 0) {
+                            Log.d("PUI", "load more is called $currentPage")
+                            postDetailViewModel.getPost(postDetailExtras.postId, currentPage)
+                        }
+                    }
+                }
+            setPaginationScrollListener(paginationScrollListener)
+        }
+    }
+
+    private fun initSwipeRefreshLayout() {
+        mSwipeRefreshLayout = binding.swipeRefreshLayout
+        mSwipeRefreshLayout.apply {
+            setColorSchemeColors(
+                //todo: change this color as per the style
+                ContextCompat.getColor(
+                    requireContext(),
+                    R.color.lm_feed_majorelle_blue
+                )
+            )
+
+            setOnRefreshListener {
+                refreshPostData()
+            }
+        }
+    }
+
+    // refreshes the whole post detail screen
+    private fun refreshPostData() {
+        mSwipeRefreshLayout.isRefreshing = true
+        binding.rvPostDetails.resetScrollListenerData()
+        fetchPostData(true)
+    }
+
+    private fun fetchData() {
+        fetchPostData()
+        checkCommentsRight()
+    }
+
+    //fetches post data to set initial data
+    private fun fetchPostData(fromRefresh: Boolean = false) {
+        if (!fromRefresh) {
+            //show progress bar
+            LMFeedProgressBarHelper.showProgress(binding.progressBar)
+        }
+
+        //if source is notification/deep link, then call initiate first and then other apis
+        if (postDetailExtras.source == LMFeedAnalytics.Source.NOTIFICATION ||
+            postDetailExtras.source == LMFeedAnalytics.Source.DEEP_LINK
+        ) {
+            //todo: ask where should we implement this
```

Comment:
please confirm whether we have added this or not.

Reply:
this is added in further PRs